### ISSUE TITLE
s905x Bluetooth QCA9377

### ISSUE THREAD

**Issue by DiogoSilva48** [ophub/amlogic-s9xxx-armbian#476]
Hi. I'm the same user as these 3 last issues ( #342 #338 #98 )which helped me a lot, thank you. Now I have this problem where the Bluetooth doesn't work, I had the same problem in LibreElec where Bluetooth doesn't work because the dtb doesn't start it.

I had the same issue with LibreElec and the devs were able to fix it by making a custom dtb that loads the kernel driver.

https://forum.libreelec.tv/thread/22564-official-le-test-images-for-amlogic-kodi-20/?pageNo=26 here's the thread where it's mentioned.

https://github.com/LibreELEC/LibreELEC.tv/blob/a75f168a876fcea98c160d96b5e15c0bf4d2f941/projects/Amlogic/devices/AMLGX/patches/linux/amlogic-0070-WIP-arm64-dts-meson-add-support-for-TBee-Box.patch here's the patch they did.

I tried the exact same dtb here but didn't boot, maybe you can make a dtb based on the regular p212 that makes my Bluetooth work? Thank you and sorry for trouble.

**Comment by ophub** [comment]
[meson-gxl-s905x-tbee.dtb.zip](https://github.com/ophub/amlogic-s9xxx-armbian/files/9300647/meson-gxl-s905x-tbee.dtb.zip)

Download this zip file and unzip it

This file adds the qca9377-bt driver on the basis of meson-gxl-s905x-p212.dts, you download it, copy it to the /boot/dtb/amlogic directory, modify the value is `meson-gxl-s905x-tbee.dtb` in /boot/uEnv.txt, restart and try , feedback the result

**Comment by DiogoSilva48** [comment]
I get this kernel panic
![image](https://user-images.githubusercontent.com/38466084/183925399-da599ecc-2e77-4a79-981c-4b5bbdefa00b.png)

**Comment by ophub** [comment]
`dmesg | grep Blue*`

**Comment by ophub** [comment]
which dtb did you originally use?

**Comment by DiogoSilva48** [comment]
> `dmesg | grep Blue*`

I can't, it panics, I can't type anything.

> which dtb did you originally use? This is the bluetooth added on meson-gxl-s905x-p212.dts

I used that exact one. Let me try again and see.

**Comment by ophub** [comment]
Which version of the kernel are you using now? 5.15.y?

**Comment by DiogoSilva48** [comment]
> Which version of the kernel are you using now? 5.15.y?

Yes, 5.15. Tried again, just switching the dtb to the other one it boots, changing again to the new one, panic.

**Comment by ophub** [comment]
[meson-gxl-s905x-tbee.dtb.zip](https://github.com/ophub/amlogic-s9xxx-armbian/files/9301041/meson-gxl-s905x-tbee.dtb.zip)


Compiled under the 5.15.59 kernel, you download this, just above is the 5.10.135 kernel

**Comment by DiogoSilva48** [comment]
<img width="576" alt="image" src="https://user-images.githubusercontent.com/38466084/183938741-4cb6c118-1615-4144-b745-d53523c793e5.png">
Yup, it's working fine now, thank you very much.

Btw, if I don't use ethernet, is there a way to remove all that delay of 1minute and a half on the "Rising Network Connections" at boot?

**Comment by ophub** [comment]
Ok, I submitted this dtb to the upstream kernel source code, it can be used later, what is the name of your box? Post a product link.

**Comment by DiogoSilva48** [comment]
> Ok, I submitted this dtb to the upstream kernel source code, it can be used later, what is the name of your box? Post a product link.

The product is called Tbee, here's the product link https://www.tbee.com/product/tbee-box/

**Comment by ophub** [comment]
Can the wireless be used after changing this dtb?

**Comment by DiogoSilva48** [comment]
Yes, Wireless works just fine, thank you!!
